### ISSUE TITLE
Foreign keys with BigSerial not working well

### ISSUE THREAD

**Issue by Abdelhadi92** [piccolo-orm/piccolo#933]
The following Tables classes didn't create same data type for primary key and foreign key:

```
class Operation(Table, tablename="operation__2"):
    id = BigSerial(primary_key=True)


class OperationTransaction(Table, tablename="operation_transaction__2"):
    id = BigSerial(primary_key=True)
    operation = ForeignKey(
        references=Operation,
        target_column="id",
        required=True,
        null=False,
        on_delete=OnDelete.cascade,
        on_update=OnUpdate.restrict
    )  
```

<img width="822" alt="Screenshot 2024-02-25 at 10 28 52 PM" src="https://github.com/piccolo-orm/piccolo/assets/19409912/4811741b-f6db-4d9e-8b6e-56ab0ae4eebd">
<img width="832" alt="Screenshot 2024-02-25 at 10 29 00 PM" src="https://github.com/piccolo-orm/piccolo/assets/19409912/36e69aa9-79c5-4fa7-b68b-c33c23795033">

**Comment by dantownsend** [comment]
OK, I think I see the problem.

We have this logic, which is setting the foreign key type to an `INTEGER` when it's a `SERIAL` or `BIGSERIAL`:

https://github.com/piccolo-orm/piccolo/blob/46e72f439ba5e9e441c57e3942622d0d6d1024bf/piccolo/columns/column_types.py#L1847-L1848

If it's `BIGSERIAL` it should be setting it to a `BIGINT` instead.

I think this will be a fairly simple fix - thanks for reporting the problem.

**Comment by dantownsend** [comment]
This should be fixed now in `piccolo==1.3.1`.
